specification/streamanalytics/resource-manager/readme.md

```diff
@@ -123,6 +168,31 @@ regenerate-manager: true
 generate-interface: true
 ```
 
+### Tag: package-2017-04-preview and java
+
+These settings apply only when `--tag=package-2017-04-preview --java` is specified on the command line.
+Please also specify `--azure-libraries-for-java-folder=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-2017-04-preview' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.streamanalytics.v2017_04_01_preview
+  output-folder: $(azure-libraries-for-java-folder)/sdk/streamanalytics/mgmt-v2017_04_01_preview
+regenerate-manager: true
+generate-interface: true
+```
+
+### Tag: package-2020-03-preview and java
```

Comment:
Is there no distinction between pure vs composite in Java?

Reply:
Similar question for the other languages

Reply:
> Is there no distinction between pure vs composite in Java?

Thank you for catching it.
There may be only one Tag to represent 2020_03_01_preview in Java or Go, `pure` or `composite`. Because they share the same namespace and path name. We think using 'package-pure-2020-03-preview'  is better here, I will update that.

For reference:

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/readme.md#tag-package-compute-only-2017-12-and-java

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/readme.go.md#tag-package-compute-2017-12-and-go

Reply:
What about the other languages?

How will SDK generation work for languages like Java or Go then if we are only doing the pure version of the API? Are you saying users using Java SDK will only be able to call cluster APIs but not streaming job APIs?

Reply:
For my understand,  Java will generate SDK based on :
```
### Java multi-api

``` yaml $(java) && $(multiapi)
batch:
  - tag: package-pure-2017-04-preview
  - tag: package-pure-2020-03-preview
```
that will include pure-2017-04-preview and pure-2020-03-preview, released package name may be like `azure-mgmt-streamanalytics`. Users using Java SDK will be able to call both cluster APIs and streaming job APIs?

***

For Go, SDK will be published in Github, and sdk code path will include api version,like:
```
azure-sdk-for-go/services/streamanalytics/mgmt/2016-03-01/streamanalytics/
azure-sdk-for-go/services/preview/streamanalytics/mgmt/2017-04-01-preview/streamanalytics/
azure-sdk-for-go/services/preview/streamanalytics/mgmt/2020-03-01-preview/streamanalytics/
```
Users using go SDK will be able to call streaming job api by importing .../mgmt/2017-04-01-preview/streamanalytics, call cluster api by importing ...mgmt/2020-03-01-preview/streamanalytics/

reference：
https://github.com/Azure/azure-sdk-for-go/tree/master/services/streamanalytics/mgmt/2016-03-01/streamanalytics

***

For ruby, I don't know about that, looks closer to java

***

For C# and ts, using tag=package-2020-03-preview, published SDK will include composite api version.

***
We can confirm that with @weidongxu-microsoft  He may give us the best practices